boot/freeldr/freeldr/miscboot.c

```diff
@@ -145,8 +145,8 @@ LoadAndBootSector(
         BiosDriveNumber = 0;
     if (!BiosDriveNumber)
     {
-        BiosDriveNumber = FrldrBootDrive;
-        PartitionNumber = FrldrBootPartition;
+        BiosDriveNumber = FrldrGetBootDrive();
+        PartitionNumber = FrldrGetBootPartition();
     }
 
 
```

Comment:
LoadAddress assignment at line 165 below uses arch-specific implementation in PC-98 case, please fix.

```
boot/freeldr/freeldr/CMakeFiles/rosload.dir/miscboot.c.obj: in function `LoadAndBootSector':
D:\test\reactos\output-MinGW-i386/../boot/freeldr/freeldr/miscboot.c:165: undefined reference to `Pc98GetBootSectorLoadAddress'
```

Spotted by @disean in the chat. https://chat.reactos.org/reactos/pl/mgu7msakjjdndmpwacptdfmiac

Reply:
Fixed it (see last commit)

Reply:
Thank you. It could probably be squashed with one of preceding commits.